### PR TITLE
Update upgrade scenario for StableConnectIdentities feature gate

### PR DESCRIPTION
**Pull request by see-quick**
### Type of change

- Enhancement / new feature

### Description

This PR adds two test cases for upgrade and downgrade. This upgrade differs from the current one in that we also use KafkaConnect and KafkaConnector in such test cases. Also, it is using a recent feature gate `StableConnectIdentities` (upgrade only). The downgrade scenario is just testing the upgrade of (Deployment -> Deployment). When `0.35.0` is ready, we can change the behaviour of downgrade only just changing the `downgrade` yaml file to use a `StableConnectIdentities` feature gate.

### Checklist

- [x] Write tests
- [x] Make sure all tests pass

### CODE REVIEW

**Review by scholzj — COMMENTED** [review]
As there seem to be no Connect upgrade tests, this probabůy doesn't do anything right now. Should it change to Draft or be closed until the Connect tests are added to Upgrade?

**Comment by PaulRMellor** [comment]
Does any consideration need to be given to downgrade?

File: systemtest/src/test/java/io/strimzi/systemtest/upgrade/AbstractUpgradeST.java [hunk 1]
```diff
@@ -57,20 +71,23 @@ public class AbstractUpgradeST extends AbstractST {
 
     private static final Logger LOGGER = LogManager.getLogger(AbstractUpgradeST.class);
 
+    protected  File dir = null;
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/upgrade/AbstractUpgradeST.java:74]:
```suggestion
    protected File dir = null;
```

File: systemtest/src/test/java/io/strimzi/systemtest/upgrade/StrimziUpgradeIsolatedST.java [hunk 2]
```diff
@@ -154,6 +161,58 @@ void testUpgradeAcrossVersionsWithNoKafkaVersion(ExtensionContext extensionConte
...
+        // Verify FileSink KafkaConnector before upgrade
+        String connectorPodName = kubeClient().listPodsByPrefixInName(testStorage.getNamespaceName(), clusterName + "-connect").get(0).getMetadata().getName();
+        KafkaConnectUtils.waitForMessagesInKafkaConnectFileSink(testStorage.getNamespaceName(), connectorPodName, Constants.DEFAULT_SINK_FILE_PATH, "\"Hello-world - 499\"");
+
+        // Upgrade CO to HEAD and wait for readiness of ClusterOperator
+        changeClusterOperator(acrossUpgradeData, testStorage.getNamespaceName(), extensionContext);
+
+        // Verify that KafkaConnect pods are rolling and KafkaConnector is ready
+        RollingUpdateUtils.waitTillComponentHasRolled(testStorage.getNamespaceName(), connectLabelSelector, 1, connectPods);
+        KafkaConnectorUtils.waitForConnectorReady(testStorage.getNamespaceName(), clusterName);
```
Comment (im-konge) [inline systemtest/src/test/java/io/strimzi/systemtest/upgrade/StrimziUpgradeIsolatedST.java:200]: You should wait for Kafka cluster rolling update as well

**Review by im-konge — COMMENTED** [review]
Two small comments.
For the downgrade scenario -> would it make sense to start with `+StableConnectIdentities` and downgrade to empty FG?

File: systemtest/src/test/java/io/strimzi/systemtest/upgrade/AbstractUpgradeST.java [hunk 3]
```diff
@@ -199,7 +219,7 @@ protected void waitForReadinessOfKafkaCluster() {
         DeploymentUtils.waitForDeploymentAndPodsReady(clusterOperator.getDeploymentNamespace(), KafkaResources.entityOperatorDeploymentName(clusterName), 1);
     }
 
-    protected void changeClusterOperator(BundleVersionModificationData versionModificationData, String namespace, ExtensionContext extensionContext) throws IOException {
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/upgrade/AbstractUpgradeST.java:202]: wrong format?

**Review by Frawless — APPROVED** [review]

**Review by im-konge — APPROVED** [review]

**Review by jankalinic — APPROVED** [review]
Nice

**Review by scholzj — COMMENTED** [review]
I think having a proper description of what the PR does would make the reviews easier now and tracking back the changes later.

**Comment by see-quick** [comment]
> I think having a proper description of what the PR does would make the reviews easier now and tracking back the changes later.

Done.

**Pull request merged**